src/main/java/com/google/devtools/build/lib/remote/GrpcRemoteCache.java

```diff
@@ -323,15 +322,19 @@ public void upload(
     if (!uploadAction) {
       return;
     }
-    retrier.execute(
-        () ->
-            acBlockingStub()
-                .updateActionResult(
-                    UpdateActionResultRequest.newBuilder()
-                        .setInstanceName(options.remoteInstanceName)
-                        .setActionDigest(actionKey.getDigest())
-                        .setActionResult(result)
-                        .build()));
+    try {
+      retrier.execute(
+          () ->
+              acBlockingStub()
+                  .updateActionResult(
+                      UpdateActionResultRequest.newBuilder()
+                          .setInstanceName(options.remoteInstanceName)
+                          .setActionDigest(actionKey.getDigest())
+                          .setActionResult(result)
+                          .build()));
+    } catch (StatusRuntimeException e) {
+      throw new IOException(e);
```

Comment:
Please use `new IOException(e.getMessage(), e)`, else `IOException.getMessage()` will contain the `StatusRuntimeException` in the message.

Reply:
Is the behavior bad? IMO, something like `IOException: StatusRuntimeException: UNAVAILABLE` is better than `IOException: UNAVAILABLE`.

I think if we want to have better error messages for specific cases, we should create more exceptions like `CacheNotFoundException` (e.g., `CacheDownException`) and produce nice error messages in the generic caching layer.

Reply:
Looking at your example I'd argue that both are equally bad :-). Let's leave it as is and fix error messages as a follow up.